File: Desktop/WinViewer/C#/WinViewer/CredentialsDialog.cs
using System.Windows.Forms;
using System.Drawing;
using System.Reflection;

namespace ActiveReports.Samples.Viewer
{
	public partial class CredentialsDialog : Form
	{
		/// <summary>
		/// Shows credentials request dialog.
		/// </summary>
		[Obfuscation(Feature = "renaming", Exclude = true)]
		/// <summary>
		/// Allows internal callers to access to user name value entered.
		/// </summary>
		public string UserName => txtUser.Text;

		/// <summary>
		/// Allows internal callers to access to password value entered.
		/// </summary>
		public string Password => txtPassword.Text;
		public CredentialsDialog(string dataSourceName, string prompt)
		{
			InitializeComponent();
			this.MaximumSize = new Size(this.Size.Width * 2, this.Size.Height);
			this.MinimumSize = this.Size;
			Text = string.Format(Text, dataSourceName);
			this.lblPrompt.Text = prompt;
			this.txtUser.KeyDown += AskCredentialsDialog_KeyDown;
			this.txtPassword.KeyDown += AskCredentialsDialog_KeyDown;
			this.KeyDown += AskCredentialsDialog_KeyDown;
		}

		private void AskCredentialsDialog_KeyDown(object sender, KeyEventArgs e)
		{
			switch (e.KeyCode)
			{
				case Keys.Enter:
					this.DialogResult = DialogResult.OK;
					break;
				case Keys.Escape:
					this.DialogResult = DialogResult.Cancel;
					break;
			}
		}
	}
}

File: Desktop/WinViewer/VB.NET/WinViewer/CredentialsDialog.vb
Imports System.Text

Public Class CredentialsDialog
	Public ReadOnly Property UserName As String
		Get
			Return txtUser.Text
		End Get
	End Property

	Public ReadOnly Property Password As String
		Get
			Return txtPassword.Text
		End Get
	End Property

	Public Sub New(dataSourceName As String, prompt As String)
		InitializeComponent()
		Text = String.Format(Text, dataSourceName)
		lblPrompt.Text = prompt
		AddHandler Me.txtUser.KeyDown, AddressOf AskCredentialsDialog_KeyDown
		AddHandler Me.txtPassword.KeyDown, AddressOf AskCredentialsDialog_KeyDown
		AddHandler Me.KeyDown, AddressOf AskCredentialsDialog_KeyDown
	End Sub

	Private Sub AskCredentialsDialog_KeyDown(sender As Object, e As KeyEventArgs)
		Select Case e.KeyCode
			Case Keys.Enter
				Me.DialogResult = DialogResult.OK
			Case Keys.Escape
				Me.DialogResult = DialogResult.Cancel
		End Select
	End Sub
End Class